source/extensions/dynamic_modules/abi.h

```diff
@@ -846,6 +987,45 @@ bool envoy_dynamic_module_callback_http_get_dynamic_metadata_string(
 void envoy_dynamic_module_callback_http_clear_route_cache(
     envoy_dynamic_module_type_http_filter_envoy_ptr filter_envoy_ptr);
 
+/**
+ * envoy_dynamic_module_callback_http_filter_get_attribute_string is called by the module to get
+ * the string attribute value. If the attribute is not accessible or the
+ * value is not a string, this returns false.
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleHttpFilter object of the
+ * corresponding HTTP filter.
+ * @param attribute_id is the ID of the attribute.
+ * @param result_buffer_ptr is the pointer to the pointer variable where the pointer to the
+ * buffer of the value will be stored.
+ * @param result_length is the pointer to the variable where the length of the buffer will be
+ * stored.
+ * @return true if the operation is successful, false otherwise.
+ *
+ * Note: currently, not all attributes are implemented.
+ */
+bool envoy_dynamic_module_callback_http_filter_get_attribute_string(
```

Comment:
note: just like the existing dynamic metadata callbacks, each function is typed vs opaque "returning some serialized buffer" as we saw in Wasm. Non-integer/string complex type can be supported later (with some non-protobuf serialization scheme) if someone requests.